hr_attendance_validation/models/hr_attendance.py

```diff
@@ -0,0 +1,90 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HrAttendance(models.Model):
+    _inherit = "hr.attendance"
+
+    is_overtime_due = fields.Boolean(
+        string="Is overtime due",
+        default=False,
+        help="What ever the overtime is due or not. "
+        "By default overtime are not due until manager validate them.",
+    )
+    validation_sheet_id = fields.Many2one(
+        "hr.attendance.validation.sheet",
+        string="Validation sheet",
+    )
+
+    @api.model
+    def create(self, *args, **kwargs):
+        attendance = super().create(*args, **kwargs)
+        if attendance._is_validated_employee_week():
+            raise ValidationError(
+                _(
+                    "Can not create new attendance for employee %s. "
+                    "Attendance validation week has been already reviewed and validated "
+                    "for the given check_in date %s"
+                )
+                % (
+                    attendance.employee_id.name,
+                    attendance.check_in.date(),
+                )
+            )
+        return attendance
+
+    def unlink(self, *args, **kwargs):
+        for record in self:
+            if record.validation_sheet_id.state == "validated":
+                raise ValidationError(
+                    _(
+                        "Can not remove this attendance (%s, %s) "
+                        "which has been already reviewed and validated."
+                    )
+                    % (
+                        record.employee_id.name,
+                        record.check_in.date(),
+                    )
+                )
+        return super().unlink(*args, **kwargs)
+
+    def write(self, *args, **kwargs):
+        for record in self:
+            if record.validation_sheet_id.state == "validated":
+                raise ValidationError(
+                    _(
+                        "Can not change this attendance (%s, %s) "
+                        "which has been already reviewed and validated."
+                    )
+                    % (
+                        record.employee_id.name,
+                        record.check_in.date(),
+                    )
+                )
+        res = super().write(*args, **kwargs)
+        for record in self:
+            if record._is_validated_employee_week():
+                raise ValidationError(
+                    _(
+                        "Can not change this attendance (%s, %s) "
+                        "which would be moved to a validated week."
+                    )
+                    % (
+                        record.employee_id.name,
+                        record.check_in.date(),
+                    )
+                )
+        return res
+
+    def _is_validated_employee_week(self):
+        validated_week = self.env["hr.attendance.validation.sheet"].search_count(
```

Comment:
this search can occur with users that are not allowed to read on that model, manage access or use super user to perform this search count.